### ISSUE TITLE
[Bug]: Zoom rubber band lags in larger window

### ISSUE THREAD

**Issue by inxp** [matplotlib/matplotlib#23968]
### Bug summary

Zoom rubber band is smooth in small window. But when I increase the size of the window the performance degrades.

Here I have stretched the window across two monitors for illustration.

### Code for reproduction

```python
import matplotlib.pyplot as plt
import numpy as np

# Data for plotting
t = np.arange(0.0, 2.0, 0.01)
s = 1 + np.sin(2 * np.pi * t)

fig, ax = plt.subplots()
ax.plot(t, s)

ax.set(xlabel='time (s)', ylabel='voltage (mV)',
       title='About as simple as it gets, folks')
ax.grid()

fig.savefig("test.png")
plt.show()
```

### Actual outcome

![2022-09-20_17-23-16](https://user-images.githubusercontent.com/6796654/191314075-7e749ebf-8acf-4004-a667-2f44ef0f4a10.gif)

### Expected outcome

![2022-09-20_17-21-24](https://user-images.githubusercontent.com/6796654/191314067-9d551537-fca1-40ae-956e-2ff64a918df4.gif)

### Operating system

Windows

### Matplotlib Version

3.6

### Matplotlib Backend

QtAgg

### Python version

3.8.10

### Installation

pip

**Comment by QuLogic** [comment]
How large is large enough to be slow?

**Comment by inxp** [comment]
Update

On further testing, I found that lag is there even when dragging legends.

So I checked `paintEvent` function in `FigureCanvasQTAgg` of `\backends\backend_qtagg.py`

The delay was mainly due to following statement in `_unmultipled_rgba8888_to_premultiplied_argb32` function of  `cbook\__init__.py`
```
argb32 = np.take(rgba8888, [2, 1, 0, 3], axis=2)
```

`_unmultipled_rgba8888_to_premultiplied_argb32` function is called by `paintEvent` function

The function call was added as part of #11845 commit.

On reverting the changes done in commits #11731 and #11845, The lag was gone. Everything is working as expected.

Am I doing something wrong?
